src/KK.DotNet.BackgroundTasks.Scheduled/SchedulerExtensions.cs

```diff
@@ -1,12 +1,99 @@
 ﻿namespace KK.DotNet.BackgroundTasks.Scheduled
 {
+    using System;
+    using Microsoft.Extensions.Configuration;
     using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.DependencyInjection.Extensions;
+    using Microsoft.Extensions.Hosting;
 
     public static class SchedulerExtensions
     {
-        public static IServiceCollection AddScheduledTask<TScheduledTask>(this IServiceCollection services)
+        public static IScheduledTaskOptions<TScheduledTask> AddScheduledTaskOptions<TScheduledTask>(
+            this IServiceCollection services,
+            IConfiguration configuration
+        )
             where TScheduledTask : class, IScheduledTask
         {
+            if (services == null) throw new ArgumentNullException(nameof(services));
+            if (configuration == null) throw new ArgumentNullException(nameof(configuration));
+
+            var options = new ScheduledTaskOptions<TScheduledTask>();
+            configuration.Bind(options);
+            services.AddSingleton<IScheduledTaskOptions<TScheduledTask>>(options);
+            return options;
+        }
+
+        public static IScheduledTaskOptions<TScheduledTask> AddScheduledTaskOptions<TScheduledTask>(
+            this IServiceCollection services,
+            IScheduledTaskOptions<TScheduledTask> options
+        )
+            where TScheduledTask : class, IScheduledTask
+        {
+            if (services == null) throw new ArgumentNullException(nameof(services));
+            if (options == null) throw new ArgumentNullException(nameof(options));
+
+            services.AddSingleton<IScheduledTaskOptions<TScheduledTask>>(options);
+            return options;
+        }
+
+        public static IScheduledTaskOptions<TScheduledTask> AddScheduledTaskOptions<TScheduledTask>(
+            this IServiceCollection services,
+            Action<IScheduledTaskOptions<TScheduledTask>> configureOptions
+        )
+            where TScheduledTask : class, IScheduledTask
+        {
+            if (services == null) throw new ArgumentNullException(nameof(services));
+            if (configureOptions == null) throw new ArgumentNullException(nameof(configureOptions));
+
+            var options = new ScheduledTaskOptions<TScheduledTask>();
+            configureOptions.Invoke(options);
+            services.AddSingleton<IScheduledTaskOptions<TScheduledTask>>(options);
+            return options;
+        }
+
+        public static IServiceCollection AddScheduledTask<TScheduledTask>(
```

Comment:
All overloads of this method use the same logic. Please merge it.

Reply:
implemented in 89f1f8a933dbf00fbcc5e1a8ba9f2653467ed73c